refresh.template.py

```diff
@@ -212,36 +215,105 @@ def _get_cpp_command_for_files(compile_action: json):
+        Also includes one entry per header, describing one way it is compiled (to work around https://github.com/clangd/clangd/issues/123).
+
+    Crucially, this de-Bazels the compile commands it takes as input, leaving
+    something clangd can understand. The result is a command that could be run from
+    the workspace root directly, with no bazel-specific compiler wrappers,
+    environment variables, etc.
+    """
 
     # Process each action from Bazelisms -> file paths and their clang commands
     # Threads instead of processes because most of the execution time is farmed out to subprocesses. No need to sidestep the GIL
     with concurrent.futures.ThreadPoolExecutor() as threadpool:
         outputs = threadpool.map(_get_cpp_command_for_files, aquery_output.actions)
 
-    bazel_workspace_dir = os.environ['BUILD_WORKSPACE_DIRECTORY'] # Set by `bazel run`. Can't call `bazel info workspace` because bazel is running us outside the workspace.
-    # Bazel gotcha warning: If you were tempted to use `bazel info execution_root` as the build working directory for compile_commands...search ImplementationReadme.md to learn why that breaks.
-
     # Dump em to stdout as compile_commands.json entries
     header_file_entries_written = set()
     for source_files, header_files, command in outputs:
         # Only emit one entry per header
         # This makes the output vastly smaller, which has been a problem for users.
-            # e.g. https://github.com/insufficiently-caffeinated/caffeine/pull/577 
+        # e.g. https://github.com/insufficiently-caffeinated/caffeine/pull/577
         # Without this, we emit an entry for each header for each time it is included, which is explosively duplicative--the same reason why C++ compilation is slow and the impetus for the new modules.
         # Revert when https://github.com/clangd/clangd/issues/123 is solved, which would remove the need to emit headers, because clangd would take on that work.
         # If https://github.com/clangd/clangd/issues/681, we'd probably want to find a way to filter to one entry per platform.
         header_files = [h for h in header_files if h not in header_file_entries_written]
         header_file_entries_written.update(header_files)
 
         for file in itertools.chain(source_files, header_files):
-            sys.stdout.write(json.dumps({
-                'file': file,
-                'command': command,
-                'directory': bazel_workspace_dir
-            }, indent=2, check_circular=False))
-            sys.stdout.write(',')
+            yield {
+                "file": file,
+                "command": command,
+                # Bazel gotcha warning: If you were tempted to use `bazel info
+                # execution_root` as the build working directory for
+                # compile_commands...search ImplementationReadme.md to learn why
+                # that breaks.
```

Comment:
I know you may disagree, but could we not hard-wrap this?
[Normally I'd just reformat rather than bother you.]

Reply:
Done